astroid/rebuilder.py

```diff
@@ -204,14 +214,73 @@ def _get_position_info(
             end_col_offset=t.end[1],
         )
 
+    def _fix_doc_node_position(self, node: NodesWithDocsType) -> None:
+        """Fix start and end position of doc nodes for Python < 3.8."""
+        if not self._data or not node.doc_node or node.lineno is None:
+            return
+        if PY38_PLUS:
+            return
+
+        lineno = node.lineno or 1  # lineno of modules is 0
+        end_range: Optional[int] = node.doc_node.lineno
```

Comment:
Just to be sure, you added the typing for clarity right? This typing should be correct by default I think?

Reply:
Yeah, I wanted to be explicit here even if it can be inferred.